### ISSUE TITLE
[Bug] Yarn 2 should run child processes with the same Node

### ISSUE THREAD

**Issue by segrey** [yarnpkg/berry#502]
**Describe the bug**

In Yarn 2 child processes are run with Node.js interpreter from `PATH`, not with the Node that is running Yarn itself.

**To Reproduce**

Yarn 2 (`yarn set version from sources`)
```
$ yarn --version
2.0.0-rc.7
$ which node
/home/segrey/.nvm/versions/node/v12.11.1/bin/node
$ /home/segrey/.nvm/versions/node/v12.11.1/bin/node /usr/share/yarn/bin/yarn.js node -e "console.log(process.version)"
v12.11.1
$ /home/segrey/.nvm/versions/node/v10.8.0/bin/node /usr/share/yarn/bin/yarn.js node -e "console.log(process.version)"
v12.11.1
```

Works correctly in Yarn 1:

```
$ yarn --version
1.19.0
$ which node
/home/segrey/.nvm/versions/node/v12.11.1/bin/node
$ /home/segrey/.nvm/versions/node/v12.11.1/bin/node /usr/share/yarn/bin/yarn.js node -e "console.log(process.version)"
yarn node v1.19.0
v12.11.1
Done in 0.04s.
$ /home/segrey/.nvm/versions/node/v10.8.0/bin/node /usr/share/yarn/bin/yarn.js node -e "console.log(process.version)"
yarn node v1.19.0
v10.8.0
Done in 0.06s.
```

**Environment if relevant (please complete the following information):**

 - OS: Linux
 - Node version v12.11.1, v10.8.0
 - Yarn version 1.19.0, 2.0.0-rc7

**Comment by arcanis** [comment]
Found the problem - the v1 has a bug where it spawns the file pointed by [`yarn-path`](https://next.yarnpkg.com/configuration/yarnrc#yarnPath) as any other process (whereas the v2 [always assumes it's a JS file](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-cli/sources/main.ts#L11), and uses `process.execPath` to spawn it).

That causes the checked-in v2 binary to be executed via the `node` binary available in the PATH instead of the one originally used to spawn the v1 binary. Will fix that on the v1 side.

**Comment by arcanis** [comment]
Will be fixed by https://github.com/yarnpkg/yarn/pull/7592

**Comment by arcanis** [comment]
Closing, the fix got released in the 1.19.1 👍

**Comment by segrey** [comment]
Great, thanks!
